### PR TITLE
For #11851: Close tab tray when last tab closed

### PR DESCRIPTION
**Pull request by sblatz**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/utils/Undo.kt [hunk 1]
```diff
@@ -63,6 +66,31 @@ fun CoroutineScope.allowUndo(
...
+        val shouldUseBottomToolbar = view.context.settings().shouldUseBottomToolbar
+        val toolbarHeight = view.context.resources
+            .getDimensionPixelSize(R.dimen.browser_toolbar_height)
+
+        snackbar.view.setPadding(
+            0,
+            0,
+            0,
+            if (
+                paddedForBottomToolbar &&
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/utils/Undo.kt:78]: Won't this break other places the Undo is used? from a quick search - bookmarks, search engine list? I suppose having unnecessary padding is better than covering the toolbar? But still weird.

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayController.kt [hunk 2]
```diff
@@ -117,6 +117,7 @@ class DefaultTabTrayController(
         }
 
         showUndoSnackbar(snackbarMessage, snapshot)
+        dismissTabTray()
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/tabtray/TabTrayController.kt:120]: Do we need this? Aren't we already calling `dismissTabTrayIfNecessary` in the `showUndoSnackbar` fun?

File: app/src/main/java/org/mozilla/fenix/utils/Undo.kt (hunk 1 above)
Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/utils/Undo.kt:78]: Ah I see it's a param we're passing in not just the bottom toolbar setting. Okey that should be fine. We should also just generally retest the Undo in the places it's used to make sure they all still work 🆗

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt [hunk 3]
```diff
@@ -185,19 +186,31 @@ class TabTrayDialogFragment : AppCompatDialogFragment() {
...
-                getString(R.string.snackbar_deleted_undo),
-                {
-                    sessionManager.add(snapshot.session, isSelected, engineSessionState = state)
-                    tabTrayView.scrollToTab(snapshot.session.id)
-                },
-                operation = { },
-                elevation = ELEVATION,
-                anchorView = snackbarAnchor
-            )
+        val isLastOpenTab = sessionManager.sessions.size == 1
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt:189]: So we only want to close the tabs tray if private and normal tabs are gone ?

**Comment by ekager** [comment]
Can we also verify the remove all tabs works in the same way for both normal and private (ie the followup from https://github.com/mozilla-mobile/fenix/issues/12383)

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayController.kt (hunk 2 above)
Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayController.kt:120]: `showUndoSnackbarForTab` is the only one that does `dismissTabTrayIfNecessary`. The one for `showUndoSnackbar` does not. I should probably rename the `forTab` one to explicitly call out that it dismisses the tab tray.

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt (hunk 3 above)
Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt:189]: Still waiting on a UX answer for this, but I'm probably going to change it to be based on the active session type (private or normal).

**Pull request merged**